cdntaxreceipts.functions.inc

```diff
@@ -1035,28 +1035,23 @@ function _receiptThankYouHeader($pdf, $pdf_variables, $receipt) {
       'sequential' => 1,
       'id' => $receipt['contact_id'],
     ]);
-    $returnProperties = ['display_name'];
-    $contactTokenValues = CRM_Utils_Token::getTokenDetails(['contact_id' => $receipt['contact_id']],
-      $returnProperties,
-      true,
-      true,
-      NULL,
-      $messageToken
-    );
-    $contactValues = $contact['values'][0];
-    if(!empty($contactTokenValues)) {
-      $contactValues = array_merge($contactValues, $contactTokenValues[0][$receipt['contact_id']]);
-    }
-    $categories = CRM_Cdntaxreceipts_Task_PDFLetterCommon::getTokenCategories();
-    $tokenHtml = CRM_Utils_Token::replaceContactTokens($newString, $contact, TRUE, $messageToken);
-    $domain = CRM_Core_BAO_Domain::getDomain();
-    $tokenHtml = CRM_Utils_Token::replaceDomainTokens($tokenHtml, $domain, TRUE, $messageToken);
-    $tokenHtml = CRM_Utils_Token::replaceContributionTokens($tokenHtml, $pdf_variables, TRUE, $messageToken);
-    if(!empty($contactTokenValues)) {
-      $tokenHtml = CRM_Utils_Token::replaceHookTokens($tokenHtml, $contactValues, $categories, TRUE);
+    
+    //CRM-1792 Replacing deprecated way of token processing with new token Processor
+    $groupedContributions = [];
+    $grouped = FALSE;
+    $realSeparator = ', ';
+    $separator = '****~~~~';
```

Comment:
What are realSeparator and separator used for? Where are they coming from?
Can you add a bit of context?

Reply:
@celinenicolas Thank you Celine for your inputs..!!  :)